Exercise_2.py

```diff
@@ -1,22 +1,69 @@
+'''
+Time Complexity :
+init(): O(1)
+push(): O(1)
+pop(): O(1)
+
+Space Complexity :
+init(): O(1)
+push(): O(1)
+pop(): O(1)
+
+Did this code successfully run on Leetcode :
+Didnt run since the identifiers used had different names and I was running late to individually rename all.
+
+Any problem you faced while coding this :
+Yes, tried implementing the stack without adding the variable "prev" in the Node class but was not successful.
+'''
```

Comment:
You can just have data and next in the Node. Have a head pointer and push the new element at the beginning of list and pop from the head element.